### DISCUSSION TITLE
HmIPW-DRAP lässt sich nicht anlernen.

### DISCUSSION THREAD

**Discussion by Sixt-Haider** [OpenCCU/OpenCCU#2531] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

ich habe RaspberryMatic als Addon auf meinem HA (läuft auf einem Intel NUC) und benutze den HmIP-RFUSB um meine HmIP Geräte damit zu verbinden. Alle Funkgeräte konnten Problemlos Angemelder werden, nur mein DRAP macht mir Probleme und wird nicht erkannt. Habe schon einiges Versucht, zuletzt beide Geräte (Intel NUC und DRAP) direkt an die Fritzbox angeschlossen, komplette Port-Freigabe für beide Geräte und sogar einen neuen DRAP gekauft um auszuschließen, dass es an meinem "alten" hängt. Das Ergebnis bleibt immer gleich, im Posteingang wird kein Gerät erkannt. Deshalb nun hier mein Hilferuf ins Forum, woran kann es noch liegen? 

Die RaspberryMatic ist auf Version 3.71.12

Hier das Protokoll:
```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.2, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:00:14.0-3, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:00:14.0-3, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .......OK
Starting rfd: .OK
Starting HMIPServer: ........OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.71.12.20231020 (raspmatic_oci_amd64)

```
Vielen Dank im Voraus für Eure Hilfe!

**Comment by jens-maus** [comment, marked as answer]
Da hättest du den Abschnitt "Known limitations" in der offiziellen Dokumentation des RaspberryMatic HA-Addon lesen sollen. Siehe hier ganz oben:

https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant

Zusammenfassend: Auf Grund von Limitationen in HomeAssistant kann die Unterstützung für die Anbindung eines `HmIP-HAP` oder `HmIP-DRAP` nur mit hilfe eines experimentellen / manuellen Patches aktuell umgesetzt werden der auch leider nach jedem Neustart von HomeAssistant oder dem RaspberryMatic Addon erneut manuell angewendet werden muss (siehe https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch)

**Reply by jens-maus** [reply]
Deshalb hätte ich für eine bessere/langfristigere Lösung dieses Problems folgenden Ratschlag:

Statt direkt HomeAssistantOS auf deinem IntelNUC zu installieren würde ich dir raten stattdessen ProxmoxVE (https://www.proxmox.com/de/proxmox-virtual-environment) als Basis-OS zu installieren und dann HomeAssistantOS sowie RaspberryMatic als separate virtuelle VMs dort drauf zu betreiben und nicht RaspberryMatic innerhalb von HomeAssistant als HA-Addon zu nutzen. Damit würdest du dir viel erleichtern.

Das würde nicht nur deine Probleme mit der Anbindung eines `HmIP-DRAP` sofort lösen, es ermöglicht dir dann auch vielfältige weitere Möglichkeiten, ein einfacheres Backup deiner kritischen Anwendungen und darüberhinaus noch viel mehr Flexibilität, z.B. noch mehr VMs drauf zu betreiben!

**Reply by Sixt-Haider** [reply]
Vielen Dank für die schnelle Hilfe! Ich habe das mit den "Known limitations" wahrgenommen, war mir aber nicht Sicher ob hier etwas anderes gemeint war. In der Allgemeinen Dokumentation wird ja schon mehr oder minder suggeriert, dass die Verwendung der HmIPW-Produkte mit dem DRAP Funktioniert oder lese ich das aus der Tabelle falsch heraus?

![Screenshot 2023-11-27 182257](https://github.com/jens-maus/RaspberryMatic/assets/140599299/571a148c-5675-4b3f-b8d2-0bc42f872160)

**Reply by Baxxy13** [reply]
Grundsätzlich funktioniert das auch. Nur eben beim Homeassistant-Addon aktuell nur mit dem experimentellen Patch und bei einer Docker-RM mit einem zusätzlichen "shim-netzwerk".

**Reply by bf8392** [reply]
Hallo, kennt jemand eine Möglichkeit diesen Patch https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch in Home Assistant zu automatisieren bzw. eine Regel zu schreiben, so dass er quasi persistent wird?

**Reply by jens-maus** [reply]
Siehe https://github.com/jens-maus/RaspberryMatic/discussions/2468#discussioncomment-8218621
